### PR TITLE
Removes cherry bombs

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
rscdel: While experimenting with floral genetic engineering, Nanotrasen botanists discovered an explosive variety of the cherry plant. Due to gross misuse, the genes used to produce this strain have been isolated and removed while Central Command decides how best to modify it.
/ :cl:

I feel as though they haven't been properly implemented and I'd like some more time to decide how they'd work and how to implement them.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/bags.dm [hunk 1]
```diff
@@ -115,6 +115,13 @@
 	can_hold = list(/obj/item/weapon/reagent_containers/food/snacks/grown,/obj/item/seeds,/obj/item/weapon/grown)
 	burn_state = FLAMMABLE
 
+/obj/item/weapon/storage/bag/plants/can_be_inserted(obj/item/W, stop_messages = 0, mob/living/user)
+	if(istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/cherry_bomb))
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/bags.dm]: bad code

**Comment by TechnoAlchemisto** [comment]
thanks for changing your pic xhuis

File: code/game/objects/items/weapons/storage/bags.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/objects/items/weapons/storage/bags.dm]: What do you want, then?

Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/bags.dm]: good code

Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/bags.dm]: In all seriousness that check is really snowflake and bad, I'm not sure what the ideal solution is though

**Comment by Lumipharon** [comment]
Wood logs don't fit in a plant bag - how does that work code wise, and could that be used for this?

**Comment by PKPenguin321** [comment]
wood logs aren't technically plants afaik

**Comment by kevinz000** [comment]
B L U E S P A C E T R A S H B A G

**Comment by Core0verload** [comment]
Don't like it at all. Maybe nerf cherry bomb to IED levels?

**Comment by Bawhoppen** [comment]
the travis isn't showing up what

File: code/game/objects/items/weapons/storage/bags.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/bags.dm]:
```
/obj/item/weapon/reagent_containers/food/snacks/grown
   ...
   var/valid_for_plant_bag = TRUE

/obj/item/weapon/reagent_containers/food/snacks/grown/cherry_bomb
   ...
   valid_for_plant_bag = FALSE
```

Comment (optimumtact) [inline code/game/objects/items/weapons/storage/bags.dm]: a var to every single grown snack just for cherry bombs?
:neutral_face:

**Comment by Bawhoppen** [comment]
>oranges closing and reopening just to trigger jordie

File: code/game/objects/items/weapons/storage/bags.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/bags.dm]: Given growns are the only thing that can go into plant bags (bar seeds maybe idk), Yes.
You're starting to sound like Erro and his washing machine, "15+ istypes is SURELY better than a var! right guys? guys..?"

Comment (optimumtact) [inline code/game/objects/items/weapons/storage/bags.dm]: actually I was thinking of either a proc or a list, but sure strawman away.

**Comment by Xhuis** [comment]
Since everyone was angry about the spaghetti code, I did a more streamlined fix. Check the OP for more details.

File: code/game/objects/items/weapons/storage/bags.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/bags.dm]: I wasn't meaning for it to be like that oranges, I meant like, just don't go down that route.
also a proc is overkill.

File: code/modules/hydroponics/grown/berries.dm [hunk 2]
```diff
@@ -154,6 +154,13 @@
 	seed = /obj/item/seeds/cherry/bomb
 	bitesize_mod = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/cherry_bomb/New()
+	..()
+	if(potency <= 50)
```
Comment (lzimann) [inline code/modules/hydroponics/grown/berries.dm]: the potency is declared in the seed and not in the grown plant itself

Comment (Xhuis) [inline code/modules/hydroponics/grown/berries.dm]: Yeah, noticed this a few minutes ago. I'll push a fix.

**Comment by Xhuis** [comment]
As of the next commit, I'll be removing cherry bombs from the mutation list pending a re-implementation when I've decided how to proceed.

**Comment by optimumtact** [comment]
Actually I closed and reopened it to trigger travis

**Comment by Jordie0608** [comment]
You can do that from the Travis site

**Comment by optimumtact** [comment]
There was no travis build for it, as in it had missed the pr completely, so I could not

**Comment by Lumipharon** [comment]
inb4 memesters mass produce these via xenobio or some shit.

**Comment by Xhuis** [comment]
If people want them badly enough that they'll start playing xenobio, they can be my guests!

**Pull request merged**